### PR TITLE
Migrate CollectionTest to Kotlin

### PR DESCRIPTION
**Pull request by oyeraghib**
## Purpose / Description
Migrate CollectionTest to Kotlin

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by Akshay0701** [comment]
```
BUILD FAILED in 1m 58s
Wrote HTML report to file:///home/runner/work/Anki-Android/Anki-Android/AnkiDroid/build/reports/lint-results-playDebug.html

> Task :AnkiDroid:lintPlayDebug FAILED
/home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/CollectionTest.kt:40: Error: Use kotlin.test.assert[Not]Null OR kotlin.test.junit.JUnitAsserter instead of JUnit in Kotlin [LegacyNullAssertionDetector from com.ichi2.anki:lint-rules]
        assertNotNull("Collection could not be opened", col)
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/CollectionTest.kt:35: Error: Variable name should not use field prefixes. [VariableNamingDetector from com.ichi2.anki:lint-rules]
    @get:Rule
    ^
2 errors, 0 warnings
```

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/CollectionTest.kt [hunk 1]
```diff
@@ -15,29 +15,28 @@
...
-@RunWith(androidx.test.ext.junit.runners.AndroidJUnit4.class)
-public class CollectionTest extends InstrumentedTest {
-
-    @Rule
-    public GrantPermissionRule mRuntimePermissionRule =
-            GrantPermissionRule.grant(Manifest.permission.WRITE_EXTERNAL_STORAGE);
+@RunWith(AndroidJUnit4::class)
+@KotlinCleanup("Fix IDE lint")
+class CollectionTest : InstrumentedTest() {
+    @get:Rule
```
Comment (Akshay0701) [inline AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/CollectionTest.kt:35]: prefix naming giving the error, change the name convention

**Review by Akshay0701 — CHANGES_REQUESTED** [review]
you missed to change the name of first commit

**Comment by david-allison** [comment]
```
> Task :AnkiDroid:lintPlayDebug FAILED
/home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/CollectionTest.kt:40: Error: Use kotlin.test.assert[Not]Null OR kotlin.test.junit.JUnitAsserter instead of JUnit in Kotlin [LegacyNullAssertionDetector from com.ichi2.anki:lint-rules]
        assertNotNull("Collection could not be opened", col)
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 errors, 0 warnings
```

**Review by david-allison — CHANGES_REQUESTED** [review]
Doesn't compile

```
> Task :AnkiDroid:deleteSingleFileInGradlee: /Users/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/CollectionTest.kt: (27, 15): Unresolved reference: test
e: /Users/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/CollectionTest.kt: (40, 9): Unresolved reference: assertNotNull
 UP-TO-DATE
> Task :AnkiDroid:jacocoPlayDebug
```

**Comment by oyeraghib** [comment]
After adding the dependency and running the test it's showing few more deprecations in other files but they are just warnings but the test is still failing.
```
> Task :AnkiDroid:compileAmazonDebugAndroidTestJavaWithJavac
D:\anki\AnkiDroid\src\androidTest\java\com\ichi2\anki\DeckPickerTest.java:83: warning: [deprecation] <T>assertThat(T,Matcher<? super T>) in Assert has been deprecated
        assertThat(getActivityInstance(), instanceOf(StudyOptionsActivity.class));
        ^
  where T is a type-variable:
    T extends Object declared in method <T>assertThat(T,Matcher<? super T>)
error: warnings found and -Werror specified
D:\anki\AnkiDroid\src\androidTest\java\com\ichi2\libanki\DBTest.java:46: warning: [deprecation] <T>assertThat(T,Matcher<? super T>) in Assert has been deprecated
        assertThat(journalMode.toLowerCase(Locale.ROOT), not(is("wal")));
        ^
  where T is a type-variable:
    T extends Object declared in method <T>assertThat(T,Matcher<? super T>)
1 error
2 warnings

> Task :AnkiDroid:compileAmazonDebugAndroidTestJavaWithJavac FAILED
> Task :AnkiDroid:deleteSingleFileInGradle UP-TO-DATE

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.4.2/userguide/command_line_interface.html#sec:command_line_warnings
42 actionable tasks: 1 executed, 1 from cache, 40 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':AnkiDroid:compileAmazonDebugAndroidTestJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 11s

```

**Comment by mikehardy** [comment]
marking as draft for queue management

**Comment by david-allison** [comment]
Feel free to mark as ready for review when changes are made.

I've just started on collection.java, so there may be conflicts

**Comment by oyeraghib** [comment]
closing as an another PR already migrated the file.

**Pull request closed without merging**